### PR TITLE
Handled Authentication Slow Query

### PR DESCRIPTION
**Pull request by antobeslie**

### CODE REVIEW

File: src/main/kotlin/com/hypto/iam/server/db/repositories/UserRepo.kt [hunk 1]
```diff
@@ -69,12 +68,10 @@ object UserRepo : BaseRepo<UsersRecord, Users, String>() {
...
         .fetchOne()
 
-    suspend fun findByAliasUsername(aliasUsername: String): UsersRecord? {
-        val conditions = DSL.or(aliasUsernames.map { it.eq(aliasUsername) })
-
+    suspend fun findByPreferredUsername(preferredUsername: String): UsersRecord? {
         return ctx("users.findByAliasUsername")
             .selectFrom(USERS)
-            .where(conditions)
+            .where(USERS.PREFERRED_USERNAME.eq(preferredUsername))
```
Comment (hari-hypto) [inline src/main/kotlin/com/hypto/iam/server/db/repositories/UserRepo.kt:74]: Using "union all" is also one of the suggested approach to query with or clause.

https://dba.stackexchange.com/questions/172907/multicolumn-indexes-and-or-clause

File: src/main/kotlin/com/hypto/iam/server/service/UsersService.kt [hunk 2]
```diff
@@ -219,8 +220,12 @@ class UsersServiceImpl : KoinComponent, UsersService {
     }
 
     override suspend fun authenticate(username: String, password: String): User {
-        val userRecord = userRepo.findByAliasUsername(username)
-            ?: throw AuthenticationException("Invalid username and password combination")
+        val userRecord = if (EMAIL_REGEX.toRegex().matches(username)) {
+            userRepo.findByEmail(username)
```
Comment (hari-hypto) [inline src/main/kotlin/com/hypto/iam/server/service/UsersService.kt:224]: Can preferred username ever match with email regex ? If so, this condition can break.

Comment (antobeslie) [reply src/main/kotlin/com/hypto/iam/server/service/UsersService.kt:224]: preferred_username accepts only Alphanumeric and Underscore. So, email can't be set as username. This way need not even query the preferred_username column.

File: src/main/kotlin/com/hypto/iam/server/db/repositories/UserRepo.kt (hunk 1 above)
Comment (antobeslie) [reply src/main/kotlin/com/hypto/iam/server/db/repositories/UserRepo.kt:74]: This way need not even query the preferred_username column. UNION can cost more as it checks both columns, merges them and then applies conditions like deleted=false, verified=true.

**Review by hari-hypto — APPROVED** [review]

**Pull request merged**